sources/packages/api/src/route-controllers/institution/institution.institutions.controller.ts

```diff
@@ -52,12 +161,358 @@ export class InstitutionInstitutionsController extends BaseController {
+        institution.id,
+        bceidUserAccount,
+        payload,
+      );
+
+    return { id: createdInstitutionUser.id };
+  }
+
+  /**
+   * Get Institution user types and roles.
+   * @returns Institution user types and roles.
+   */
+  @IsInstitutionAdmin()
+  @Get("user-types-roles")
+  async getUserTypesAndRoles(): Promise<InstitutionUserTypeAndRoleAPIOutDTO> {
+    return this.institutionService.getUserTypesAndRoles();
+  }
+
+  /**
+   * Get institution user by user name(guid).
+   * @param userName
+   * @returns
+   */
+  @ApiNotFoundResponse({
+    description: "User not found.",
+  })
+  @ApiUnprocessableEntityResponse({
+    description: "User not active.",
+  })
+  @IsInstitutionAdmin()
+  @Get("user/:userName")
+  async getInstitutionUserByUserName(
+    @Param("userName") userName: string,
+  ): Promise<InstitutionUserAPIOutDTO> {
+    // Get institutionUser
+    const institutionUser =
+      await this.institutionService.getInstitutionUserByUserName(userName);
+
+    if (!institutionUser) {
+      throw new NotFoundException("User not found.");
+    }
+    // disabled users details can't be edited
+    if (!institutionUser.user.isActive) {
+      throw new UnprocessableEntityException("Not an Active User.");
+    }
+    return {
+      id: institutionUser.id,
+      user: {
+        firstName: institutionUser.user?.firstName,
+        lastName: institutionUser.user?.lastName,
+        userName: institutionUser.user?.userName,
+        isActive: institutionUser.user?.isActive,
+        id: institutionUser.user?.id,
+        email: institutionUser.user.email,
+      },
+      authorizations: institutionUser.authorizations?.map((el) => {
+        return {
+          location: {
+            name: el.location?.name,
+            data: el.location?.data,
+            id: el.location?.id,
+          },
+          authType: { type: el.authType?.type, role: el.authType?.role },
+        };
+      }),
+    };
+  }
+
+  /**
+   * Update the permissions of institution user.
+   * @param token
+   * @param userName
+   * @param payload
+   */
+  @ApiNotFoundResponse({
+    description: "User to be updated not found.",
+  })
+  @ApiUnprocessableEntityResponse({
+    description:
+      "User to be updated doesn't belong to institution of logged in user.",
+  })
+  @IsInstitutionAdmin()
+  @Patch("user/:userName")
+  async UpdateInstitutionUserWithAuth(
+    @UserToken() token: IInstitutionUserToken,
+    @Param("userName") userName: string,
+    @Body() payload: InstitutionUserPermissionAPIInDTO,
+  ): Promise<void> {
+    // Check its a active user
+    const institutionUser =
+      await this.institutionService.getInstitutionUserByUserName(userName);
+
+    if (!institutionUser) {
+      throw new NotFoundException("User to be updated not found");
+    }
+
+    // checking if user belong to logged-in users institution
+    if (institutionUser.institution.id !== token.authorizations.institutionId) {
```

Comment:
I believe that this is the authorization check that is missing in the above method `getInstitutionUserByUserName`.